aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -8,7 +8,16 @@
 #if HAS_CUDNN_V8()
 
 #include <ATen/cudnn/cudnn-wrapper.h>
+
+#pragma GCC diagnostic push
```

Comment:
This looks like a repeated pattern, can one use something like `C10_CLANG_DIAGNOSTIC_PUSH`?

Reply:
yes we can, but I think we'll both be wrong when GCC 9 comes out and we use `-Wsuggest-override` in GCC.

basically we want logic that works in both Clang and GCC with the following algorithm:

```
push
if warning exists "-Wsuggest-override"
  suppress warning
include header
pop
```

I don't think we can test that the warning exists in GCC, but we can probably suppress the unknown warning.

In Clang it appears to flag an unknown warning with `-Wunknown-warning-option` and in GCC it warns with `-Wpragmas`.

We can probably just suppress those warnings in the stack as well!

Let me whip something up.

Reply:
@malfet please take another look, I think this is about as simple as it can get since we can't include a file in a macro.